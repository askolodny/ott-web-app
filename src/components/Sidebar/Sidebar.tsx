import React, { Fragment } from 'react';
import classNames from 'classnames';

import MenuButton from '../../components/MenuButton/MenuButton';
import IconButton from '../../components/IconButton/IconButton';
import Close from '../../icons/Close';

import styles from './Sidebar.module.scss';

type SidebarProps = {
  isOpen: boolean;
  onClose: () => void;
  playlistMenuItems: JSX.Element[];
};

const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, playlistMenuItems }) => {
  return (
    <Fragment>
      <div
        className={classNames(styles.backdrop, {
          [styles.visible]: isOpen,
        })}
        onClick={onClose}
      />
      <div
        className={classNames(styles.sidebar, {
          [styles.open]: isOpen,
        })}
      >
        <div className={styles.heading}>
          <IconButton onClick={onClose} aria-label="close menu">
            <Close />
          </IconButton>
        </div>
        <nav className={styles.group} onClick={onClose}>
          <MenuButton label="Home" to="/" />
          {playlistMenuItems}
          <hr className={styles.divider} />
          <MenuButton label="Settings" to="/u" />
        </nav>
      </div>
    </Fragment>
  );
};

export default Sidebar;
