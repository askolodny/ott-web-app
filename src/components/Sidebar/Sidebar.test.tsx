import React from 'react';

import { render } from '../../testUtils';
import MenuButton from '../MenuButton/MenuButton';

import Sidebar from './Sidebar';

describe('<SideBar />', () => {
  const playlistMenuItems = [<MenuButton key="key" label="Playlist" to="/p/vdsvb" />];
  test('renders sideBar', () => {
    const { container } = render(<Sidebar isOpen={true} onClose={() => ''} playlistMenuItems={playlistMenuItems} />);

    expect(container).toMatchSnapshot();
  });
});
