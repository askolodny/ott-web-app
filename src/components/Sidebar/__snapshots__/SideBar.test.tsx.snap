// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> renders sideBar 1`] = `
<div>
  <div
    class="backdrop visible"
  />
  <div
    class="sidebar open"
  >
    <div
      class="heading"
    >
      <div
        aria-label="close menu"
        class="iconButton"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="icon"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </div>
    </div>
    <nav
      class="group"
    >
      <a
        aria-current="page"
        class="menuButton active"
        href="/"
      >
        <span>
          Home
        </span>
      </a>
      <a
        aria-current="page"
        class="menuButton active"
        href="/"
      >
        <span>
          Home
        </span>
      </a>
      <hr
        class="divider"
      />
      <a
        class="menuButton"
        href="/u"
      >
        <span>
          Settings
        </span>
      </a>
    </nav>
  </div>
</div>
`;
